src/models/events.rs

```diff
@@ -0,0 +1,294 @@
+pub mod payload;
+
+use self::payload::{CreateEventPayload, EventPayload, PushEventPayload};
+use chrono::{DateTime, Utc};
+use reqwest::Url;
+use serde::{de::Error, Deserialize, Serialize};
+
+/// A GitHub event.
+#[derive(Debug, Clone, PartialEq, Serialize)]
+#[non_exhaustive]
+pub struct Event {
+    pub id: String,
+    #[serde(rename = "type")]
+    pub typ: EventType,
```

Comment:
I find using the exact name as the JSON response more intuitive, even if you have to use the rare `r#` syntax.
```suggestion
    pub r#type: EventType,
```